src/adapter/src/coord.rs

```diff
@@ -2551,9 +2555,8 @@ impl Coordinator {
                 }
 
                 // If something is _really_ slow, print a trace id for debugging, if OTEL is enabled.
-                let trace_id_threshold = Duration::from_secs(5).min(prometheus_threshold * 25);
-                if duration > trace_id_threshold && otel_context.is_valid() {
-                    let trace_id = otel_context.trace_id();
+                if duration > warn_threshold {
+                    let trace_id = otel_context.is_valid().then(|| otel_context.trace_id());
```

Comment:
Why is setting this to `None` in the otel off case useful?

Reply:
It's not useful, but it's easier to trace `None` than it is to have a separate branch. For example, I think the alternative would be something like:
```
let trace_id = otel_context.is_valid().then(|| otel_context.trace_id());
let msg = "very slow coordinator message";
if let Some(trace_id) = trace_id {
    tracing::warn!(?msg_kind, ?trace_id, ?duration, msg);
} else {
    tracing::warn!(?msg_kind, ?duration, msg);
}
```
More than happy to make this change if you'd like!